adapter-jdbc/src/main/java/io/nosqlbench/adapter/jdbc/JDBCSpace.java

```diff
@@ -0,0 +1,184 @@
+            }
+            hikariConfig.setPassword(password.get());
+        } else {
+            if (user.isPresent()) {
+                throw new OpConfigError("Both user and password options are required. Only user is supplied in this case.");
+            }
+        }
+
+        Optional<Boolean> ssl = cfg.getOptional(Boolean.class, "ssl");
+        hikariConfig.addDataSourceProperty("ssl", ssl.orElse(false));
+
+        Optional<String> sslMode = cfg.getOptional("sslmode");
+        if (sslMode.isPresent()) {
+            hikariConfig.addDataSourceProperty("sslmode", sslMode.get());
+        } else {
+            hikariConfig.addDataSourceProperty("sslmode", "prefer");
+        }
+
+        Optional<String> sslCert = cfg.getOptional("sslcert");
+        if (sslCert.isPresent()) {
+            hikariConfig.addDataSourceProperty("sslcert", sslCert.get());
+        } /*else if(sslMode.isPresent() && (!"disable".equalsIgnoreCase(sslMode.get()) || !"allow".equalsIgnoreCase(sslMode.get())) || !"prefer".equalsIgnoreCase(sslMode.get())) {
+            throw new OpConfigError("When sslmode is true, sslcert should be provided.");
+        }*/
+
+        Optional<String> sslRootCert = cfg.getOptional("sslrootcert");
+        if (sslRootCert.isPresent()) {
+            hikariConfig.addDataSourceProperty("sslrootcert", sslRootCert.get());
+        }
+
+        hikariConfig.addDataSourceProperty("applicationName", cfg.get("applicationName"));
+        hikariConfig.addDataSourceProperty("rewriteBatchedInserts", cfg.getOrDefault("rewriteBatchedInserts", true));
+
+        // We're managing the auto-commit behavior of connections ourselves and hence disabling the auto-commit.
+        //Optional<Boolean> autoCommit = cfg.getOptional(Boolean.class, "autoCommit");
+        hikariConfig.setAutoCommit(false);
+
+        hikariConfig.setMaximumPoolSize(Integer.parseInt(cfg.get("maximumPoolSize")));
+        hikariConfig.setKeepaliveTime(Integer.parseInt(cfg.get("keepaliveTime")));
+        hikariConfig.setMaximumPoolSize(Integer.parseInt(cfg.get("maximumPoolSize")));
+
+        HikariDataSource hds = new HikariDataSource(hikariConfig);
+        try {
+            this.connection = hds.getConnection();
+            // We're taking an opinionated approach here and managing the commit ourselves.
+            this.getConnection().setAutoCommit(false);
+        } catch (Exception ex) {
+            String exp = "Exception occurred while attempting to create a connection using the HikariDataSource";
+            logger.error(exp, ex);
+            throw new RuntimeException(exp, ex);
+        }
+
+        return hds;
+    }
+
+    public static NBConfigModel getConfigModel() {
+        return ConfigModel.of(JDBCSpace.class)
+            .add(Param.defaultTo("url", "jdbc:postgresql:/").setDescription("The connection URL used to connect to the DBMS. Defaults to 'jdbc:postgresql:/'"))
```

Comment:
You could wrap these lines with `.setDescription(...)` on separate lines for readability.

Reply:
Done